### PR TITLE
(User roles) Add scopes to routes

### PR DESCRIPTION
**Pull request by tumbledwyer**
Change access to routes from roles to scopes.

Dependent on https://github.com/opencrvs/opencrvs-core/pull/7741

### CODE REVIEW

File: packages/client/src/App.tsx [hunk 1]
```diff
@@ -307,11 +384,9 @@ export function App(props: IAppProps) {
                                             />
                                             <ProtectedRoute
                                               exact
-                                              path={routes.VERIFY_COLLECTOR}
-                                              component={VerifyCollector}
-                                            />
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:312]: We don't want to remove this route now do we?

File: packages/client/src/App.tsx [hunk 2]
```diff
@@ -307,11 +384,9 @@ export function App(props: IAppProps) {
...
-                                            <ProtectedRoute
-                                              exact
+                                              scopes={[
+                                                SCOPES.RECORD_REGISTRATION_REQUEST_CORRECTION
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:388]:
```suggestion
                                                SCOPES.RECORD_REGISTRATION_CORRECT
```

File: packages/client/src/App.tsx [hunk 3]
```diff
@@ -327,13 +402,19 @@ export function App(props: IAppProps) {
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.RECORD_REGISTRATION_CORRECT
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:406]:
```suggestion
                                                SCOPES.RECORD_REGISTRATION_REQUEST_CORRECTION
```

File: packages/client/src/App.tsx [hunk 4]
```diff
@@ -327,13 +402,19 @@ export function App(props: IAppProps) {
...
+                                              ]}
                                               path={
                                                 routes.CERTIFICATE_CORRECTION
                                               }
                                               component={CorrectionForm}
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.RECORD_REGISTRATION_CORRECT
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:416]:
```suggestion
                                                SCOPES.RECORD_REGISTRATION_REQUEST_CORRECTION
```
I know this is a bit confusing but CERTIFICATE_CORRECTION_REVIEW is actually a part of the request correction form

File: packages/client/src/App.tsx [hunk 5]
```diff
@@ -346,122 +427,120 @@ export function App(props: IAppProps) {
...
-                                              //   SystemRoleType.LocalRegistrar,
-                                              //   SystemRoleType.LocalSystemAdmin,
-                                              //   SystemRoleType.NationalSystemAdmin,
-                                              //   SystemRoleType.PerformanceManagement
-                                              // ]}
+                                              scopes={[
+                                                SCOPES.ORGANISATION_READ,
+                                                SCOPES.ORGANISATION_READ_LOCATIONS,
+                                                SCOPES.ORGANISATION_READ_LOCATIONS_MY_OFFICE
+                                              ]}
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:434]: https://github.com/opencrvs/opencrvs-core/pull/7790/files#diff-ec2dc392bcf26c2a7c1a34631db8a5f5d220596d2e986f8959acaa22494fcb78R353-R363

File: packages/client/src/App.tsx [hunk 6]
```diff
@@ -346,122 +427,120 @@ export function App(props: IAppProps) {
...
                                               path={routes.SYSTEM_LIST}
                                               component={SystemList}
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.USER_CREATE,
+                                                SCOPES.USER_CREATE_MY_JURISDICTION,
+                                                SCOPES.USER_READ_MY_OFFICE
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:460]:
```suggestion
                                                SCOPES.USER_CREATE,
                                                SCOPES.USER_CREATE_MY_JURISDICTION
```

File: packages/client/src/App.tsx [hunk 7]
```diff
@@ -346,122 +427,120 @@ export function App(props: IAppProps) {
...
+                                              ]}
                                               path={routes.CREATE_USER}
                                               component={CreateNewUser}
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.USER_CREATE,
+                                                SCOPES.USER_CREATE_MY_JURISDICTION,
+                                                SCOPES.USER_READ_MY_OFFICE
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:470]:
```suggestion
                                                SCOPES.USER_CREATE,
                                                SCOPES.USER_CREATE_MY_JURISDICTION
```

File: packages/client/src/App.tsx [hunk 8]
```diff
@@ -346,122 +427,120 @@ export function App(props: IAppProps) {
...
                                                 routes.CREATE_USER_ON_LOCATION
                                               }
                                               component={CreateNewUser}
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.USER_CREATE,
+                                                SCOPES.USER_CREATE_MY_JURISDICTION,
+                                                SCOPES.USER_READ_MY_OFFICE
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:482]:
```suggestion
                                                SCOPES.USER_CREATE,
                                                SCOPES.USER_CREATE_MY_JURISDICTION
```

File: packages/client/src/App.tsx [hunk 9]
```diff
@@ -346,122 +427,120 @@ export function App(props: IAppProps) {
...
+                                              ]}
                                               path={routes.CREATE_USER_SECTION}
                                               component={CreateNewUser}
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.USER_CREATE,
+                                                SCOPES.USER_CREATE_MY_JURISDICTION,
+                                                SCOPES.USER_READ_MY_OFFICE
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:492]:
```suggestion
                                                SCOPES.USER_CREATE,
                                                SCOPES.USER_CREATE_MY_JURISDICTION
```

File: packages/client/src/App.tsx [hunk 10]
```diff
@@ -346,122 +427,120 @@ export function App(props: IAppProps) {
...
+                                              ]}
                                               path={routes.REVIEW_USER_FORM}
                                               component={CreateNewUser}
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.USER_CREATE,
+                                                SCOPES.USER_CREATE_MY_JURISDICTION,
+                                                SCOPES.USER_READ_MY_OFFICE
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:502]:
```suggestion
                                                SCOPES.USER_CREATE,
                                                SCOPES.USER_CREATE_MY_JURISDICTION
```

File: packages/client/src/App.tsx [hunk 11]
```diff
@@ -346,122 +427,120 @@ export function App(props: IAppProps) {
...
                                               exact
-                                              // roles={[
-                                              //   SystemRoleType.RegistrationAgent,
-                                              //   SystemRoleType.LocalRegistrar,
-                                              //   SystemRoleType.LocalSystemAdmin,
-                                              //   SystemRoleType.NationalSystemAdmin,
-                                              //   SystemRoleType.PerformanceManagement,
-                                              //   SystemRoleType.NationalRegistrar
-                                              // ]}
+                                              scopes={[SCOPES.PERFORMANCE_READ]}
```
Comment (Zangetsu101) [inline packages/client/src/App.tsx:509]: https://github.com/opencrvs/opencrvs-core/pull/7790/files#diff-ec2dc392bcf26c2a7c1a34631db8a5f5d220596d2e986f8959acaa22494fcb78R418-R421

Comment (Zangetsu101) [reply packages/client/src/App.tsx:509]: Better to sync the performance & organisation ones from https://github.com/opencrvs/opencrvs-core/pull/7790/files

File: packages/client/src/hooks/useAuthorization.ts [hunk 12]
```diff
@@ -18,9 +18,11 @@ export function usePermissions() {
   const userPrimaryOffice = useSelector(getUserDetails)?.primaryOffice
 
   const hasScopes = (neededScopes: Scope[]) =>
+    neededScopes.length === 0 ||
     neededScopes.every((scope) => userScopes?.includes(scope))
```
Comment (Zangetsu101) [inline packages/client/src/hooks/useAuthorization.ts:22]: `every` returns true for an empty array

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/App.tsx (hunk 1 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: Thanks, good catch

File: packages/client/src/App.tsx (hunk 2 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: Done

File: packages/client/src/App.tsx (hunk 3 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: Done

File: packages/client/src/App.tsx (hunk 6 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: Hmmm... repeated copy paste error there

File: packages/client/src/App.tsx (hunk 5 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: +1

File: packages/client/src/App.tsx (hunk 7 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: :+1:

File: packages/client/src/App.tsx (hunk 8 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: :+1:

File: packages/client/src/App.tsx (hunk 9 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: :+1:

File: packages/client/src/App.tsx (hunk 11 above)
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: Done

File: packages/client/src/hooks/useAuthorization.ts (hunk 12 above)
Comment (tumbledwyer) [reply packages/client/src/hooks/useAuthorization.ts:22]: Thanks, done

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

File: packages/client/src/App.tsx [hunk 13]
```diff
@@ -307,11 +384,17 @@ export function App(props: IAppProps) {
                                             />
                                             <ProtectedRoute
                                               exact
+                                              scopes={[
+                                                SCOPES.RECORD_REGISTRATION_CORRECT
```
[the remark this answers is not in the document]
Comment (tumbledwyer) [reply packages/client/src/App.tsx:388]: So these are the print scopes, which scope or combination or scopes should it be?

`RECORD_PRINT_RECORDS`
`RECORD_PRINT_RECORDS_SUPPORTING_DOCUMENTS`
`RECORD_PRINT_ISSUE_CERTIFIED_COPIES`
`RECORD_BULK_PRINT_CERTIFIED_COPIES`

[the remark this answers is not in the document]
Comment (Zangetsu101) [reply packages/client/src/App.tsx:388]: hm the `RECORD_PRINT_RECORDS` should suffice here

[the remark this answers is not in the document]
Comment (tumbledwyer) [reply packages/client/src/App.tsx]: Thanks, done

**Pull request merged**